internal/provider/common.go

```diff
@@ -0,0 +1,147 @@
+package provider
+
+import (
+	"net/http"
+
+	"github.com/go-openapi/swag"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/featuresupport"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+)
+
+func isPlatformBM(platform *models.Platform) bool {
+	return platform != nil && *platform.Type == models.PlatformTypeBaremetal
+}
+
+func isPlatformNone(platform *models.Platform) bool {
+	return platform != nil && *platform.Type == models.PlatformTypeNone
+}
+
+func isClusterPlatformNone(cluster *common.Cluster) bool {
+	return cluster != nil && isPlatformNone(cluster.Platform)
+}
+
+func isClusterPlatformBM(cluster *common.Cluster) bool {
+	return cluster != nil && isPlatformBM(cluster.Platform)
+}
+
+func checkPlatformWrongParamsInput(platform *models.Platform, userManagedNetworking *bool, cluster *common.Cluster) error {
+	if platform != nil && userManagedNetworking != nil {
+		userManagedNetworkingStatus := "enabled"
+		if !*userManagedNetworking {
+			userManagedNetworkingStatus = "disabled"
+		}
+
+		if (!*userManagedNetworking && isPlatformNone(platform)) || (*userManagedNetworking && isPlatformBM(platform)) {
+			return common.NewApiError(http.StatusBadRequest, errors.Errorf("Can't set %s platform with user-managed-networking %s", *platform.Type, userManagedNetworkingStatus))
+		}
+	}
+
+	// If current cluster platform is different than baremetal/none, and we want to set the cluster platform to one
+	// of those platforms, that might cause the cluster to be in wrong state (baremetal + umn enabled, none + umn disabled)
+	// In those cases return bed request
+	if cluster != nil && !isClusterPlatformBM(cluster) && !isClusterPlatformNone(cluster) {
+		if platform != nil && isPlatformBM(platform) && swag.BoolValue(cluster.UserManagedNetworking) {
+			return common.NewApiError(http.StatusBadRequest, errors.Errorf("Can't set %s platform with user-managed-networking enabled", *platform.Type))
+		}
+
+		if platform != nil && isPlatformNone(platform) && !swag.BoolValue(cluster.UserManagedNetworking) {
+			return common.NewApiError(http.StatusBadRequest, errors.Errorf("Can't set %s platform with user-managed-networking disabled", *platform.Type))
+		}
+	}
+
+	return nil
+}
+
+func doesPlatformDifferentThanBaremetalOrNone(platform *models.Platform, cluster *common.Cluster) bool {
+	if platform != nil && !isPlatformBM(platform) && !isPlatformNone(platform) {
+		return true
+	}
+
+	if platform == nil && !isClusterPlatformBM(cluster) && !isClusterPlatformNone(cluster) {
+		return true
+	}
```

Comment:
I think you can remove this part, it seems that this scenario is already covered in  `GetActualUpdateClusterPlatformParams` (lines 82 and 91)